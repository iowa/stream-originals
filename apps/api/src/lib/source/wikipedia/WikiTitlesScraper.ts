import * as cheerio from "cheerio";
import type { Streamer, Title, TitleDraft } from "@repo/common";
import { WikiTitlesTable } from "./WikiTitlesTable.js";

export class WikiTitlesScraper {

  constructor(
    private readonly wikiTitlesTable: WikiTitlesTable = new WikiTitlesTable(),
  ) {

  }

  async findTitles(
    $: cheerio.CheerioAPI,
    streamer: Streamer,
  ): Promise<TitleDraft[]> {
    const titles: TitleDraft[] = [];
    $("table").each((_, table) => {
      const tableHeaders = $(table).find("th");
      if (
        tableHeaders.first().text().trim() === "Title" &&
        (tableHeaders.eq(2).text().trim() === "Premiere" ||
          tableHeaders.eq(2).text().trim() === "Release")
      ) {
        $(table)
        .find("tbody tr:has(td)")
        .each((_, row) => {
          const title = this.wikiTitlesTable.parseRow(cheerio.load(row).html());
          titles.push({
            ...title,
            streamer: streamer,
          });
        });
      }
    });
    return titles;
  }
}
