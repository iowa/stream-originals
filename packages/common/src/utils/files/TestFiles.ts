import fs from "node:fs";
import path from "node:path";

export class TestFiles {
  static loadJson(dirname: string, fileName: string) {
    return JSON.parse(fs.readFileSync(path.join(dirname, fileName), "utf8"));
  }

  static load(dirname: string, fileName: string) {
    return fs.readFileSync(path.join(dirname, fileName), "utf8");
  }
}
