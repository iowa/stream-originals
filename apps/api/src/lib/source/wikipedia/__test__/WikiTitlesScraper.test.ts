import { describe, expect, it } from "vitest";
import { dirname } from "node:path";
import { fileURLToPath } from "node:url";
import * as cheerio from "cheerio";
import { TestFiles } from "../../../utils/files/TestFiles.js";
import { WikiTitlesScraper } from "../WikiTitlesScraper.js";

describe("WikiTitlesScraper", () => {
  it("findTitles", async () => {
    const html = TestFiles.load(
      dirname(fileURLToPath(import.meta.url)),
      "/data/original_programming.html",
    );
    const $ = cheerio.load(html);
    const series = await WikiTitlesScraper.findTitles($, "appleTV+");

    expect(series).toMatchSnapshot("applePlus_titles.html");
  });
});
