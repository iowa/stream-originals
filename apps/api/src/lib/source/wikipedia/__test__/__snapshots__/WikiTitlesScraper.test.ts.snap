// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WikiTitlesScraper > findTitles > applePlus_titles.html 1`] = `
[
  {
    "name": "For All Mankind",
    "premiere": "2019-11-01",
    "streamer": "appleTV+",
  },
  {
    "name": "The Morning Show",
    "premiere": "2019-11-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Foundation",
    "premiere": "2021-09-24",
    "streamer": "appleTV+",
  },
  {
    "name": "Invasion",
    "premiere": "2021-10-22",
    "streamer": "appleTV+",
  },
  {
    "name": "Severance",
    "premiere": "2022-02-18",
    "streamer": "appleTV+",
  },
  {
    "name": "Pachinko",
    "premiere": "2022-03-25",
    "streamer": "appleTV+",
  },
  {
    "name": "Slow Horses",
    "premiere": "2022-04-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Surface",
    "premiere": "2022-07-29",
    "streamer": "appleTV+",
  },
  {
    "name": "Bad Sisters",
    "premiere": "2022-08-19",
    "streamer": "appleTV+",
  },
  {
    "name": "The Last Thing He Told Me",
    "premiere": "2023-04-14",
    "streamer": "appleTV+",
  },
  {
    "name": "Silo",
    "premiere": "2023-05-05",
    "streamer": "appleTV+",
  },
  {
    "name": "Hijack",
    "premiere": "2023-06-28",
    "streamer": "appleTV+",
  },
  {
    "name": "The Buccaneers",
    "premiere": "2023-11-08",
    "streamer": "appleTV+",
  },
  {
    "name": "Monarch: Legacy of Monsters",
    "premiere": "2023-11-17",
    "streamer": "appleTV+",
  },
  {
    "name": "Criminal Record",
    "premiere": "2024-01-10",
    "streamer": "appleTV+",
  },
  {
    "name": "The New Look",
    "premiere": "2024-02-14",
    "streamer": "appleTV+",
  },
  {
    "name": "Sugar",
    "premiere": "2024-04-05",
    "streamer": "appleTV+",
  },
  {
    "name": "Dark Matter",
    "premiere": "2024-05-08",
    "streamer": "appleTV+",
  },
  {
    "name": "Presumed Innocent",
    "premiere": "2024-06-12",
    "streamer": "appleTV+",
  },
  {
    "name": "Your Friends & Neighbors",
    "premiere": "2025-04-11",
    "streamer": "appleTV+",
  },
  {
    "name": "Chief of War",
    "premiere": "2025-08-01",
    "streamer": "appleTV+",
  },
  {
    "name": "The Savant",
    "premiere": "2025-09-26",
    "streamer": "appleTV+",
  },
  {
    "name": "The Last Frontier",
    "premiere": "2025-10-10",
    "streamer": "appleTV+",
  },
  {
    "name": "Down Cemetery Road",
    "premiere": "2025-10-29",
    "streamer": "appleTV+",
  },
  {
    "name": "Pluribus",
    "premiere": "2025-11-07",
    "streamer": "appleTV+",
  },
  {
    "name": "Trying",
    "premiere": "2020-05-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Ted Lasso",
    "premiere": "2020-08-14",
    "streamer": "appleTV+",
  },
  {
    "name": "Loot",
    "premiere": "2022-06-24",
    "streamer": "appleTV+",
  },
  {
    "name": "Shrinking",
    "premiere": "2023-01-27",
    "streamer": "appleTV+",
  },
  {
    "name": "Platonic",
    "premiere": "2023-05-24",
    "streamer": "appleTV+",
  },
  {
    "name": "Palm Royale",
    "premiere": "2024-03-20",
    "streamer": "appleTV+",
  },
  {
    "name": "Bad Monkey",
    "premiere": "2024-08-14",
    "streamer": "appleTV+",
  },
  {
    "name": "The Studio",
    "premiere": "2025-03-26",
    "streamer": "appleTV+",
  },
  {
    "name": "Government Cheese",
    "premiere": "2025-04-16",
    "streamer": "appleTV+",
  },
  {
    "name": "Murderbot",
    "premiere": "2025-05-16",
    "streamer": "appleTV+",
  },
  {
    "name": "Stick",
    "premiere": "2025-06-04",
    "streamer": "appleTV+",
  },
  {
    "name": "Yo Gabba GabbaLand!",
    "premiere": "2024-08-09",
    "streamer": "appleTV+",
  },
  {
    "name": "Stillwater",
    "premiere": "2020-12-04",
    "streamer": "appleTV+",
  },
  {
    "name": "Sago Mini Friends",
    "premiere": "2022-09-16",
    "streamer": "appleTV+",
  },
  {
    "name": "Shape Island",
    "premiere": "2023-01-20",
    "streamer": "appleTV+",
  },
  {
    "name": "Eva the Owlet",
    "premiere": "2023-03-31",
    "streamer": "appleTV+",
  },
  {
    "name": "Curses!",
    "premiere": "2023-10-27",
    "streamer": "appleTV+",
  },
  {
    "name": "Camp Snoopy",
    "premiere": "2024-06-14",
    "streamer": "appleTV+",
  },
  {
    "name": "WondLa",
    "premiere": "2024-06-28",
    "streamer": "appleTV+",
  },
  {
    "name": "Wonder Pets: In the City",
    "premiere": "2024-12-13",
    "streamer": "appleTV+",
  },
  {
    "name": "Goldie",
    "premiere": "2025-02-14",
    "streamer": "appleTV+",
  },
  {
    "name": "Bearbrick",
    "premiere": "2025-03-21",
    "streamer": "appleTV+",
  },
  {
    "name": "Not a Box",
    "premiere": "2025-06-13",
    "streamer": "appleTV+",
  },
  {
    "name": "The Sisters Grimm",
    "premiere": "2025-10-03",
    "streamer": "appleTV+",
  },
  {
    "name": "Women in Blue",
    "premiere": "2024-07-31",
    "streamer": "appleTV+",
  },
  {
    "name": "La Maison",
    "premiere": "2024-09-20",
    "streamer": "appleTV+",
  },
  {
    "name": "Midnight Family",
    "premiere": "2024-09-25",
    "streamer": "appleTV+",
  },
  {
    "name": "Where's Wanda?",
    "premiere": "2024-10-02",
    "streamer": "appleTV+",
  },
  {
    "name": "You Would Do It Too",
    "premiere": "2024-10-30",
    "streamer": "appleTV+",
  },
  {
    "name": "Love You to Death",
    "premiere": "2025-02-05",
    "streamer": "appleTV+",
  },
  {
    "name": "Carême",
    "premiere": "2025-04-30",
    "streamer": "appleTV+",
  },
  {
    "name": "Prehistoric Planet",
    "premiere": "2022-05-23",
    "streamer": "appleTV+",
  },
  {
    "name": "The Reluctant Traveler",
    "premiere": "2023-02-24",
    "streamer": "appleTV+",
  },
  {
    "name": "The Wild Ones",
    "premiere": "2025-07-11",
    "streamer": "appleTV+",
  },
  {
    "name": "Mr. Scorsese",
    "premiere": "2025-10-17",
    "streamer": "appleTV+",
  },
  {
    "name": "KPopped",
    "premiere": "2025-08-29",
    "streamer": "appleTV+",
  },
  {
    "name": "Countdown to First Pitch",
    "premiere": "2022-04-07",
    "streamer": "appleTV+",
  },
  {
    "name": "MLB Daily Recap",
    "premiere": "2022-04-07",
    "streamer": "appleTV+",
  },
  {
    "name": "Friday Night Baseball",
    "premiere": "2022-04-08",
    "streamer": "appleTV+",
  },
  {
    "name": "Margo's Got Money Troubles",
    "premiere": "2026-01-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Lucky",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "12 12 12",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Cape Fear",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "The Dispatcher",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Imperfect Women",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Neuromancer",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Star City",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "The Wanted Man",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Berlin Noir",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "The Dealer",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "The Off Weeks",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Untitled Joona Linna series",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Wild Things",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Brothers",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Widow's Bay",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "The Choir",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "The Husbands",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "I'm Glad My Mom Died",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Maximum Pleasure Guaranteed",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Prodigies",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "À L'ombre des Forêts",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Endangered Planet",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Knife Edge: Chasing Michelin Stars",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "Untitled UConn Huskies women's basketball docuseries",
    "premiere": undefined,
    "streamer": "appleTV+",
  },
  {
    "name": "See",
    "premiere": "2019-11-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Servant",
    "premiere": "2019-11-28",
    "streamer": "appleTV+",
  },
  {
    "name": "Truth Be Told",
    "premiere": "2019-12-06",
    "streamer": "appleTV+",
  },
  {
    "name": "Amazing Stories",
    "premiere": "2020-03-06",
    "streamer": "appleTV+",
  },
  {
    "name": "Home Before Dark",
    "premiere": "2020-04-03",
    "streamer": "appleTV+",
  },
  {
    "name": "Defending Jacob",
    "premiere": "2020-04-24",
    "streamer": "appleTV+",
  },
  {
    "name": "Little Voice",
    "premiere": "2020-07-10",
    "streamer": "appleTV+",
  },
  {
    "name": "The Mosquito Coast",
    "premiere": "2021-04-30",
    "streamer": "appleTV+",
  },
  {
    "name": "Lisey's Story",
    "premiere": "2021-06-04",
    "streamer": "appleTV+",
  },
  {
    "name": "Swagger",
    "premiere": "2021-10-29",
    "streamer": "appleTV+",
  },
  {
    "name": "The Shrink Next Door",
    "premiere": "2021-11-12",
    "streamer": "appleTV+",
  },
  {
    "name": "Suspicion",
    "premiere": "2022-02-04",
    "streamer": "appleTV+",
  },
  {
    "name": "The Last Days of Ptolemy Grey",
    "premiere": "2022-03-11",
    "streamer": "appleTV+",
  },
  {
    "name": "WeCrashed",
    "premiere": "2022-03-18",
    "streamer": "appleTV+",
  },
  {
    "name": "Roar",
    "premiere": "2022-04-15",
    "streamer": "appleTV+",
  },
  {
    "name": "Shining Girls",
    "premiere": "2022-04-29",
    "streamer": "appleTV+",
  },
  {
    "name": "The Essex Serpent",
    "premiere": "2022-05-13",
    "streamer": "appleTV+",
  },
  {
    "name": "Black Bird",
    "premiere": "2022-07-08",
    "streamer": "appleTV+",
  },
  {
    "name": "Five Days at Memorial",
    "premiere": "2022-08-12",
    "streamer": "appleTV+",
  },
  {
    "name": "Shantaram",
    "premiere": "2022-10-14",
    "streamer": "appleTV+",
  },
  {
    "name": "Echo 3",
    "premiere": "2022-11-23",
    "streamer": "appleTV+",
  },
  {
    "name": "Dear Edward",
    "premiere": "2023-02-03",
    "streamer": "appleTV+",
  },
  {
    "name": "Hello Tomorrow!",
    "premiere": "2023-02-17",
    "streamer": "appleTV+",
  },
  {
    "name": "Extrapolations",
    "premiere": "2023-03-17",
    "streamer": "appleTV+",
  },
  {
    "name": "City on Fire",
    "premiere": "2023-05-12",
    "streamer": "appleTV+",
  },
  {
    "name": "The Crowded Room",
    "premiere": "2023-06-09",
    "streamer": "appleTV+",
  },
  {
    "name": "The Changeling",
    "premiere": "2023-09-08",
    "streamer": "appleTV+",
  },
  {
    "name": "Lessons in Chemistry",
    "premiere": "2023-10-13",
    "streamer": "appleTV+",
  },
  {
    "name": "Masters of the Air",
    "premiere": "2024-01-26",
    "streamer": "appleTV+",
  },
  {
    "name": "Constellation",
    "premiere": "2024-02-21",
    "streamer": "appleTV+",
  },
  {
    "name": "Manhunt",
    "premiere": "2024-03-15",
    "streamer": "appleTV+",
  },
  {
    "name": "Franklin",
    "premiere": "2024-04-12",
    "streamer": "appleTV+",
  },
  {
    "name": "The Big Cigar",
    "premiere": "2024-05-17",
    "streamer": "appleTV+",
  },
  {
    "name": "Lady in the Lake",
    "premiere": "2024-07-19",
    "streamer": "appleTV+",
  },
  {
    "name": "Disclaimer",
    "premiere": "2024-10-11",
    "streamer": "appleTV+",
  },
  {
    "name": "Before",
    "premiere": "2024-10-25",
    "streamer": "appleTV+",
  },
  {
    "name": "Prime Target",
    "premiere": "2025-01-22",
    "streamer": "appleTV+",
  },
  {
    "name": "Dope Thief",
    "premiere": "2025-03-14",
    "streamer": "appleTV+",
  },
  {
    "name": "Smoke",
    "premiere": "2025-06-27",
    "streamer": "appleTV+",
  },
  {
    "name": "Dickinson",
    "premiere": "2019-11-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Little America",
    "premiere": "2020-01-17",
    "streamer": "appleTV+",
  },
  {
    "name": "Mythic Quest",
    "premiere": "2020-02-07",
    "streamer": "appleTV+",
  },
  {
    "name": "Physical",
    "premiere": "2021-06-18",
    "streamer": "appleTV+",
  },
  {
    "name": "Schmigadoon!",
    "premiere": "2021-07-16",
    "streamer": "appleTV+",
  },
  {
    "name": "Mr. Corman",
    "premiere": "2021-08-06",
    "streamer": "appleTV+",
  },
  {
    "name": "Acapulco",
    "premiere": "2021-10-08",
    "streamer": "appleTV+",
  },
  {
    "name": "The Afterparty",
    "premiere": "2022-01-28",
    "streamer": "appleTV+",
  },
  {
    "name": "The Big Door Prize",
    "premiere": "2023-03-29",
    "streamer": "appleTV+",
  },
  {
    "name": "High Desert",
    "premiere": "2023-05-17",
    "streamer": "appleTV+",
  },
  {
    "name": "Still Up",
    "premiere": "2023-09-22",
    "streamer": "appleTV+",
  },
  {
    "name": "The Completely Made-Up Adventures of Dick Turpin",
    "premiere": "2024-03-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Land of Women",
    "premiere": "2024-06-26",
    "streamer": "appleTV+",
  },
  {
    "name": "Sunny",
    "premiere": "2024-07-10",
    "streamer": "appleTV+",
  },
  {
    "name": "Time Bandits",
    "premiere": "2024-07-24",
    "streamer": "appleTV+",
  },
  {
    "name": "Side Quest",
    "premiere": "2025-03-26",
    "streamer": "appleTV+",
  },
  {
    "name": "Ghostwriter",
    "premiere": "2019-11-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Helpsters",
    "premiere": "2019-11-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Fraggle Rock: Rock On!",
    "premiere": "2020-04-21",
    "streamer": "appleTV+",
  },
  {
    "name": "Helpsters Help You",
    "premiere": "2020-04-24",
    "streamer": "appleTV+",
  },
  {
    "name": "Puppy Place",
    "premiere": "2021-10-15",
    "streamer": "appleTV+",
  },
  {
    "name": "Hello, Jack! The Kindness Show",
    "premiere": "2021-11-05",
    "streamer": "appleTV+",
  },
  {
    "name": "Fraggle Rock: Back to the Rock",
    "premiere": "2022-01-21",
    "streamer": "appleTV+",
  },
  {
    "name": "Lovely Little Farm",
    "premiere": "2022-06-10",
    "streamer": "appleTV+",
  },
  {
    "name": "Best Foot Forward",
    "premiere": "2022-07-22",
    "streamer": "appleTV+",
  },
  {
    "name": "Amber Brown",
    "premiere": "2022-07-29",
    "streamer": "appleTV+",
  },
  {
    "name": "Surfside Girls",
    "premiere": "2022-08-19",
    "streamer": "appleTV+",
  },
  {
    "name": "Life by Ella",
    "premiere": "2022-09-02",
    "streamer": "appleTV+",
  },
  {
    "name": "Slumberkins",
    "premiere": "2022-11-04",
    "streamer": "appleTV+",
  },
  {
    "name": "Circuit Breakers",
    "premiere": "2022-11-11",
    "streamer": "appleTV+",
  },
  {
    "name": "Jane",
    "premiere": "2023-04-14",
    "streamer": "appleTV+",
  },
  {
    "name": "Me",
    "premiere": "2024-07-12",
    "streamer": "appleTV+",
  },
  {
    "name": "Central Park",
    "premiere": "2020-05-29",
    "streamer": "appleTV+",
  },
  {
    "name": "Strange Planet",
    "premiere": "2023-08-09",
    "streamer": "appleTV+",
  },
  {
    "name": "Snoopy in Space",
    "premiere": "2019-11-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Doug Unplugs",
    "premiere": "2020-11-13",
    "streamer": "appleTV+",
  },
  {
    "name": "The Snoopy Show",
    "premiere": "2021-02-05",
    "streamer": "appleTV+",
  },
  {
    "name": "Wolfboy and the Everything Factory",
    "premiere": "2021-09-24",
    "streamer": "appleTV+",
  },
  {
    "name": "Get Rolling with Otis",
    "premiere": "2021-10-08",
    "streamer": "appleTV+",
  },
  {
    "name": "Harriet the Spy",
    "premiere": "2021-11-19",
    "streamer": "appleTV+",
  },
  {
    "name": "El Deafo",
    "premiere": "2022-01-07",
    "streamer": "appleTV+",
  },
  {
    "name": "Pretzel and the Puppies",
    "premiere": "2022-02-11",
    "streamer": "appleTV+",
  },
  {
    "name": "Pinecone & Pony",
    "premiere": "2022-04-08",
    "streamer": "appleTV+",
  },
  {
    "name": "Duck & Goose",
    "premiere": "2022-07-08",
    "streamer": "appleTV+",
  },
  {
    "name": "Interrupting Chicken",
    "premiere": "2022-11-18",
    "streamer": "appleTV+",
  },
  {
    "name": "Frog and Toad",
    "premiere": "2023-04-28",
    "streamer": "appleTV+",
  },
  {
    "name": "Dr. Brain",
    "premiere": "2021-11-03",
    "streamer": "appleTV+",
  },
  {
    "name": "Now and Then",
    "premiere": "2022-05-20",
    "streamer": "appleTV+",
  },
  {
    "name": "Liaison",
    "premiere": "2023-02-24",
    "streamer": "appleTV+",
  },
  {
    "name": "Visible: Out on Television",
    "premiere": "2020-02-14",
    "streamer": "appleTV+",
  },
  {
    "name": "Home",
    "premiere": "2020-04-17",
    "streamer": "appleTV+",
  },
  {
    "name": "Dear...",
    "premiere": "2020-06-05",
    "streamer": "appleTV+",
  },
  {
    "name": "Greatness Code",
    "premiere": "2020-07-10",
    "streamer": "appleTV+",
  },
  {
    "name": "Long Way Up",
    "premiere": "2020-09-18",
    "streamer": "appleTV+",
  },
  {
    "name": "Tiny World",
    "premiere": "2020-10-02",
    "streamer": "appleTV+",
  },
  {
    "name": "Becoming You",
    "premiere": "2020-11-13",
    "streamer": "appleTV+",
  },
  {
    "name": "Earth at Night in Color",
    "premiere": "2020-12-04",
    "streamer": "appleTV+",
  },
  {
    "name": "1971: The Year That Music Changed Everything",
    "premiere": "2021-05-21",
    "streamer": "appleTV+",
  },
  {
    "name": "The Me You Can't See",
    "premiere": "2021-05-21",
    "streamer": "appleTV+",
  },
  {
    "name": "Watch the Sound with Mark Ronson",
    "premiere": "2021-07-30",
    "streamer": "appleTV+",
  },
  {
    "name": "The Line",
    "premiere": "2021-11-19",
    "streamer": "appleTV+",
  },
  {
    "name": "Lincoln's Dilemma",
    "premiere": "2022-02-18",
    "streamer": "appleTV+",
  },
  {
    "name": "The Long Game: Bigger Than Basketball",
    "premiere": "2022-04-22",
    "streamer": "appleTV+",
  },
  {
    "name": "They Call Me Magic",
    "premiere": "2022-04-22",
    "streamer": "appleTV+",
  },
  {
    "name": "Make or Break",
    "premiere": "2022-04-29",
    "streamer": "appleTV+",
  },
  {
    "name": "The Big Conn",
    "premiere": "2022-05-06",
    "streamer": "appleTV+",
  },
  {
    "name": "Gutsy",
    "premiere": "2022-09-09",
    "streamer": "appleTV+",
  },
  {
    "name": "Super League: The War for Football",
    "premiere": "2023-01-13",
    "streamer": "appleTV+",
  },
  {
    "name": "Real Madrid: Until the End",
    "premiere": "2023-03-10",
    "streamer": "appleTV+",
  },
  {
    "name": "Monster Factory",
    "premiere": "2023-03-17",
    "streamer": "appleTV+",
  },
  {
    "name": "Boom! Boom! The World vs. Boris Becker",
    "premiere": "2023-04-07",
    "streamer": "appleTV+",
  },
  {
    "name": "Big Beasts",
    "premiere": "2023-04-21",
    "streamer": "appleTV+",
  },
  {
    "name": "Wanted: The Escape of Carlos Ghosn",
    "premiere": "2023-08-25",
    "streamer": "appleTV+",
  },
  {
    "name": "The Super Models",
    "premiere": "2023-09-20",
    "streamer": "appleTV+",
  },
  {
    "name": "Messi Meets America",
    "premiere": "2023-10-11",
    "streamer": "appleTV+",
  },
  {
    "name": "The Enfield Poltergeist",
    "premiere": "2023-10-27",
    "streamer": "appleTV+",
  },
  {
    "name": "John Lennon: Murder Without a Trial",
    "premiere": "2023-12-06",
    "streamer": "appleTV+",
  },
  {
    "name": "The Dynasty: New England Patriots",
    "premiere": "2024-02-16",
    "streamer": "appleTV+",
  },
  {
    "name": "Messi's World Cup: The Rise of a Legend",
    "premiere": "2024-02-21",
    "streamer": "appleTV+",
  },
  {
    "name": "Earthsounds",
    "premiere": "2024-02-23",
    "streamer": "appleTV+",
  },
  {
    "name": "STEVE! (martin) a documentary in 2 pieces",
    "premiere": "2024-03-29",
    "streamer": "appleTV+",
  },
  {
    "name": "Hollywood Con Queen",
    "premiere": "2024-05-08",
    "streamer": "appleTV+",
  },
  {
    "name": "Omnivore",
    "premiere": "2024-07-19",
    "streamer": "appleTV+",
  },
  {
    "name": "Cowboy Cartel",
    "premiere": "2024-08-02",
    "streamer": "appleTV+",
  },
  {
    "name": "K-Pop Idols",
    "premiere": "2024-08-30",
    "streamer": "appleTV+",
  },
  {
    "name": "Vietnam: The War That Changed America",
    "premiere": "2025-01-31",
    "streamer": "appleTV+",
  },
  {
    "name": "Onside: Major League Soccer",
    "premiere": "2025-02-21",
    "streamer": "appleTV+",
  },
  {
    "name": "Fight for Glory: 2024 World Series",
    "premiere": "2025-03-28",
    "streamer": "appleTV+",
  },
  {
    "name": "Number One on the Call Sheet",
    "premiere": "2025-03-28",
    "streamer": "appleTV+",
  },
  {
    "name": "Long Way Home",
    "premiere": "2025-05-09",
    "streamer": "appleTV+",
  },
  {
    "name": "My Kind of Country",
    "premiere": "2023-03-24",
    "streamer": "appleTV+",
  },
  {
    "name": "Oprah's Book Club",
    "premiere": "2019-11-01",
    "streamer": "appleTV+",
  },
  {
    "name": "Oprah Talks COVID-19",
    "premiere": "2020-03-21",
    "streamer": "appleTV+",
  },
  {
    "name": "The Oprah Conversation",
    "premiere": "2020-07-30",
    "streamer": "appleTV+",
  },
  {
    "name": "The Problem with Jon Stewart",
    "premiere": "2021-09-30",
    "streamer": "appleTV+",
  },
  {
    "name": "Snoopy Presents: For Auld Lang Syne",
    "premiere": "2021-12-10",
    "streamer": "appleTV+",
  },
  {
    "name": "Snoopy Presents: It's the Small Things, Charlie Brown",
    "premiere": "2022-04-15",
    "streamer": "appleTV+",
  },
  {
    "name": "Snoopy Presents: To Mom (and Dad), With Love",
    "premiere": "2022-05-06",
    "streamer": "appleTV+",
  },
  {
    "name": "Snoopy Presents: Lucy's School",
    "premiere": "2022-08-12",
    "streamer": "appleTV+",
  },
  {
    "name": "Snoopy Presents: One-of-a-Kind Marcie",
    "premiere": "2023-08-18",
    "streamer": "appleTV+",
  },
  {
    "name": "Snoopy Presents: Welcome Home, Franklin",
    "premiere": "2024-02-16",
    "streamer": "appleTV+",
  },
  {
    "name": "A Carpool Karaoke Christmas",
    "premiere": "2024-12-16",
    "streamer": "appleTV+",
  },
  {
    "name": "Snoopy Presents: A Summer Musical",
    "premiere": "2025-08-15",
    "streamer": "appleTV+",
  },
]
`;
